umh-core/pkg/config/manager.go

```diff
@@ -319,39 +353,105 @@ func (m *FileConfigManager) GetConfig(ctx context.Context, tick uint64) (FullCon
 		m.cacheMu.RUnlock()
 		return cfg, nil
 	}
+
+	// File has changed - check if we should start background refresh
+	currentCacheConfig := m.cacheConfig.Clone()
+	hasCache := !m.cacheModTime.IsZero()
 	m.cacheMu.RUnlock()
+
 	// ---------- SLOW PATH (file changed) ----------
+	// Check if a refresh is already in progress
+	m.refreshMu.Lock()
+	refreshInProgress := m.refreshInProgress
+	if !refreshInProgress && hasCache {
+		// Start background refresh only if we have a cached config to return
+		m.refreshInProgress = true
+		go m.backgroundRefresh(info.ModTime())
+	}
+	m.refreshMu.Unlock()
```

Comment:
Double check the usage of the mutexe here.In background refresh you also have the mutex